### ISSUE TITLE
Homework 2 (Functions)

### ISSUE THREAD

**Issue by cmurray3** [IE-555/fall2023#4]
Homework 2 is now posted:
- https://github.com/IE-555/fall2023/blob/main/Homework/HW_2_functions/hw_2_functions.md 

If you have questions about the assignment, please pose them by replying to this issue below.  Emailed homework questions will not be answered.

**Comment by Huize-Hou** [comment]
For question 4, how to deal with the first input (input `['abcde']`` and return 1)? I always get error of invalid syntax.

**Comment by abidaisl** [comment]
For question 2, I'm stuck on creating the function that will not take the string, the function that I created only takes integers but when user input is other than an integer the function gives an error.

**Comment by cmurray3** [comment]
> For question 4, how to deal with the first input (input `['abcde']`` and return 1)? I always get error of invalid syntax.

The input should be the list `['abcde']`, without the excess left quote marks.  For example, `count_the_elements(['abcde'])`
I have corrected the typo in the homework description.

**Comment by cmurray3** [comment]
> For question 2, I'm stuck on creating the function that will not take the string, the function that I created only takes integers but when user input is other than an integer the function gives an error.

This question is related to an example we did in class from the intro notes, where we found the min/max numeric values in a list.

**Comment by ggazetta** [comment]
Homework 2 submission https://classroom.github.com/a/5car_o8K

**Comment by ggazetta** [comment]
We have a link for the assignment:  https://classroom.github.com/a/fdmXg_gd
The deadline is 11:59 PM on Thursday, you will lose write access to the repository after that, so make sure the code is uploaded until then.

Also, please use this Issue for questions you might have!

**Comment by yusufihsantokel** [comment]
I have a question about question 5: 
we are asked to return a python **set** as output. however, when i check the error code i received after uploading the project, i realised that the "expected output" parts are python lists:

![Capture](https://github.com/IE-555/fall2023/assets/143470485/b4dcc198-667d-4c48-9f64-f593c7300824)

as you can see I have sets as outputs; however considered wrong...
am i missing something?

**Comment by ggazetta** [comment]
> I have a question about question 5: we are asked to return a python **set** as output. however, when i check the error code i received after uploading the project, i realised that the "expected output" parts are python lists:
> 
> ![Capture](https://user-images.githubusercontent.com/143470485/272149636-b4dcc198-667d-4c48-9f64-f593c7300824.JPG)
> 
> as you can see I have sets as outputs; however considered wrong... am i missing something?

No, you are right. We will consider that correct, so consider 10 points added to the grade you saw there.

**Comment by lohithub** [comment]
Hi,
 
For the Question 4, I believe I have written the code correctly since in the Spyder its showing correct output for each input but here in the repo after execution its throwing wrong and giving some random number which is not expected as an output when we run in any other IDE or platform. Not even one test case is passing, May I please know the reason for it?  attaching the Screenshot for the reference! I already reuploaded the file 2 times but this is not getting resolved! Please help me with the same!!


<img width="1432" alt="Screenshot 2023-10-04 at 2 16 47 AM" src="https://github.com/IE-555/fall2023/assets/143578326/a81b4c2a-93ef-45fa-8a3a-190d55c8decb">
<img width="1432" alt="Screenshot 2023-10-04 at 2 14 02 AM" src="https://github.com/IE-555/fall2023/assets/143578326/1692b2ce-942a-4060-9636-c992710bc05a">

Thanks,
Lohith K

**Comment by cmurray3** [comment]
> Hi,
> 
> For the Question 4, I believe I have written the code correctly since in the Spyder its showing correct output for each input but here in the repo after execution its throwing wrong and giving some random number which is not expected as an output when we run in any other IDE or platform. Not even one test case is passing, May I please know the reason for it? attaching the Screenshot for the reference! I already reuploaded the file 2 times but this is not getting resolved! Please help me with the same!!
> 
> <img alt="Screenshot 2023-10-04 at 2 16 47 AM" width="1432" src="https://user-images.githubusercontent.com/143578326/272479678-a81b4c2a-93ef-45fa-8a3a-190d55c8decb.png"> <img alt="Screenshot 2023-10-04 at 2 14 02 AM" width="1432" src="https://user-images.githubusercontent.com/143578326/272479709-1692b2ce-942a-4060-9636-c992710bc05a.png">
> Thanks, Lohith K

Hi, Lohith,

Without knowing what you've written in your `count_the_elements` function, I can't say what is wrong in your code.  However, when running your code, you should not be using the `runfile` function.  Instead, you should `import myFunctions` and then call `myFunctions.count_the_elements([1, [2, 3]])` (or some other input arguments for testing).  

Also, make sure you're not using `global` variables, as these will create issues (which is one reason we haven't discussed them in class).

**Comment by lohithub** [comment]
oh! got it sir, will try modifying the logic and I will let you know. Thank you so much

**Comment by suyasharlimar** [comment]
> > I have a question about question 5: we are asked to return a python **set** as output. however, when i check the error code i received after uploading the project, i realised that the "expected output" parts are python lists:
> > ![Capture](https://user-images.githubusercontent.com/143470485/272149636-b4dcc198-667d-4c48-9f64-f593c7300824.JPG)
> > as you can see I have sets as outputs; however considered wrong... am i missing something?
> 
> No, you are right. We will consider that correct, so consider 10 points added to the grade you saw there.

First my output was in form of set, then accoeding to test case i changed code to give output in form of list and i got all test cases passed. Do i need to change code once again beacuse my output is in form list?

**Comment by ggazetta** [comment]
**NEW assignment submission link** to correct assertion error with question 5.

https://classroom.github.com/a/aG3ER9p7

**Comment by lohithub** [comment]
> oh! got it sir, will try modifying the logic and I will let you know. Thank you so much

Yes sir, after modifying the logic from global variable to local count variable then it worked. Thank you for letting me know

**Comment by ggazetta** [comment]
> > > I have a question about question 5: we are asked to return a python **set** as output. however, when i check the error code i received after uploading the project, i realised that the "expected output" parts are python lists:
> > > ![Capture](https://user-images.githubusercontent.com/143470485/272149636-b4dcc198-667d-4c48-9f64-f593c7300824.JPG)
> > > as you can see I have sets as outputs; however considered wrong... am i missing something?
> > 
> > 
> > No, you are right. We will consider that correct, so consider 10 points added to the grade you saw there.
> 
> First my output was in form of set, then accoeding to test case i changed code to give output in form of list and i got all test cases passed. Do i need to change code once again beacuse my output is in form list?

I dont see your repository in the new homework, I posted a new link above that corrects that issue. In that one, question 5 will only accept a set. Accept the homework and submit your code that matches the homework description.

**Comment by sairohithkalasani** [comment]
Hi, can anyone confirm whether the deadline is extended or not for homework 2?

**Comment by abidaisl** [comment]
Yes , it’s due on October 5th 11:59
